poker-app/components/PracticeButtonController.js

```diff
@@ -10,60 +10,92 @@ export default class PracticeButtonController extends Component {
             raises: 0,
             tag: "",
             tags: [],
-            gamesArray : []
+            gamesArray: []
         };
     };
 
-    populateGames(){
-        storageController.retrieveData().then((res)=> {
+    async populateGames() {
+        await storageController.retrieveData().then((res) => {
             let pastGames = JSON.parse(res);
-            this.setState({
-                gamesArray: pastGames
+            console.log("populate function");
+            let arrayOfgames = [];
+            pastGames.games.forEach(game => {
+                arrayOfgames.push(game);
+                console.log(game)
             })
+            console.log("TAKE A LOOK")
+            console.log(arrayOfgames)
+            console.log(res)
+            console.log(pastGames)
+
+            this.setState({
+                gamesArray: arrayOfgames
+            });
+
+        }).catch((error) => {
+            alert("populate error");
+            throw error;
         })
     }
 
-    componentDidMount(){
-        this.populateGames();
-        console.log("LOOK UNDER");
-        console.log(this.state.gamesArray)
+
+
+
+    componentDidMount() {
+        this.populateGames().then(() => {
+            console.log("LOOK UNDER");
+            console.log(this.state.gamesArray)
+        })
     }
 
-    
-    saveToTags(tag){
+
+    saveToTags(tag) {
         let tagsArray = this.state.tags;
         tagsArray.push(tag);
         this.setState({
             tags: tagsArray
         })
     };
-    
-    toBeSaved(){
-        let gameObj =  {calls: this.state.calls,
-                folds: this.state.folds,
-                raises: this.state.raises,
-                tags: this.state.tags}
-        let arr = this.state.gamesArray;
-        arr.push(gameObj);
-        return arr; 
+
+
+
+    toBeSaved() {
+        let date = new Date();
+        let gamesObj = {
+            date: date.getDate(),
+            time: date.getTime(),
+            calls: this.state.calls,
```

Comment:
you can consider having state.currentGame and under currentGame have folds/raises/calls. Then you add currentGame to gamesArray.

Reply:
okay, that's a good idea ill might try doing that